src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -244,19 +249,35 @@ public static long countBulkFileLoadModelsFromPath(
       ObjectMapper objectMapper,
       IngestRequestModel ingestRequest,
       String cloudEncapsulationId,
-      List<String> fileRefColumnNames,
+      List<Column> fileRefColumns,
       List<String> errors) {
     return IngestUtils.getJsonNodesStreamFromFile(
             cloudFileReader, objectMapper, ingestRequest, cloudEncapsulationId, errors)
         .flatMap(
             node ->
-                fileRefColumnNames.stream()
-                    .map(node::get)
-                    .filter(n -> n != null && n.isObject())
+                fileRefColumns.stream()
+                    .map(c -> node.get(c.getName()))
+                    .filter(n -> n != null && (n.isObject() || n.isArray()))
+                    .flatMap(
+                        n -> {
+                          if (n.isObject()) {
+                            return Stream.of(n);
+                          } else {
+                            return StreamSupport.stream(
+                                Spliterators.spliteratorUnknownSize(n.iterator(), 0), false);
+                          }
+                        })
```

Comment:
It's a minor thing but I think it's clearer to check the node type once instead of in both the `filter()` and the `map()`.
```suggestion
                    .filter(Objects::nonNull)
                    .flatMap(
                        n -> {
                          if (n.isObject()) {
                            return Stream.of(n);
                          } else if (n.isArray()) {
                            return StreamSupport.stream(
                                Spliterators.spliteratorUnknownSize(n.iterator(), 0), false);
                          } else {
                             return Stream.empty();
                          }
                        })
```